Lib/test/test_bytes.py

```diff
@@ -772,8 +772,31 @@ def check(fmt, vals, result):
         check(b'%i%b %*.*b', (10, b'3', 5, 3, b'abc',), b'103   abc')
         check(b'%c', b'a', b'a')
 
-        self.assertRaisesRegex(TypeError, '%i format: a real number is required, not complex', operator.mod, '%i', 2j)
-        self.assertRaisesRegex(TypeError, '%d format: a real number is required, not complex', operator.mod, '%d', 2j)
+        class PseudoFloat:
+            def __init__(self, value):
+                self.value = float(value)
+            def __int__(self):
+                return int(self.value)
+
+        pi = PseudoFloat(3.1415)
+
+        exceptions_params = [
+            ('%x format: an integer is required, not float', '%x', 3.14),
+            ('%X format: an integer is required, not float', '%X', 2.11),
+            ('%o format: an integer is required, not float', '%o', 1.79),
+            ('%x format: an integer is required, not PseudoFloat', '%x', pi),
+            ('%x format: an integer is required, not complex', '%x', 3j),
+            ('%X format: an integer is required, not complex', '%X', 2j),
+            ('%o format: an integer is required, not complex', '%o', 1j),
+            ('%u format: a real number is required, not complex', '%u', 3j),
+            ('%i format: a real number is required, not complex', '%i', 2j),
+            ('%d format: a real number is required, not complex', '%d', 2j),
+            (r'%c requires an int or a unicode character, not .*\.PseudoFloat',
+                '%c', pi),
+        ]
+
+        for msg, format_str, value in exceptions_params:
+            self.assertRaisesRegex(TypeError, msg, operator.mod, format_str, value)
```

Comment:
I'm sorry, I didn't think that it would now exceed 80 chars here...

```suggestion
            with self.assertRaisesRegex(TypeError, msg):
                operator.mod(format_str, value)
```

Reply:
oh i didn't even notice you changed the operator. I only noticed the format